system/src/roller/Roller2D20.mjs

```diff
@@ -532,3 +548,126 @@ export class Roller2D20 {
 		return { whisper, blind };
 	}
 }
+export class reroll extends Dialog {
+	constructor(
+		actordata,
+		numOfDice,
+		rolltype,
+		dialogData={},
+		options={}
+	) {
+		super(dialogData, options);
+		this.actor = actordata;
+		this.numOfDice = numOfDice;
+		this.rolltype = rolltype;
+	}
+
+	activateListeners(html) {
+		super.activateListeners(html);
+		html.on("change", ".spend-luck", event =>  {
+			let usedluckpoints = 0;
+			if (this.rolltype === 6) {
+				usedluckpoints = Math.ceil(parseInt($(event.currentTarget).val())/ 3);
+			}
+			else {
+				usedluckpoints = parseInt($(event.currentTarget).val());
+			}
+			const nonSpendLuck = html.find(".non-spend-luck");
+			const nonSpendLucknumofdice = parseInt($(nonSpendLuck).val());
+			nonSpendLuck.empty();
+			for (let i = 0; i <= this.numOfDice - usedluckpoints; i++) {
+				const adjustedValue = Math.max(0, i);
+				const selected = i === 0 ? "selected" : "";
+				nonSpendLuck.append(`<option value="${adjustedValue}" ${selected}>${adjustedValue}</option>`);
+				nonSpendLuck[0].selectedIndex = nonSpendLucknumofdice;
+			}
+			const avaliableluck = this.actor.system.luckPoints;
+			if (usedluckpoints > avaliableluck) {
+				const part1 = game.i18n.localize("FALLOUT.UI.Not_Enough");
+				const part2 = game.i18n.localize("FALLOUT.AbilityLuc");
+				const part3 = game.i18n.localize("FALLOUT.TEMPLATES.Points");
+				ui.notifications.warn(`${part1} ${part2} ${part3}`);
+				html.find(".spend-luck")[0].selectedIndex = 0;
+				html.find(".spend-luck").trigger("change");
+
+			}
+		});
+		html.on("change", ".non-spend-luck", event => {
+			const freedicetoreroll = parseInt($(event.currentTarget).val());
+			const SpendLuck = html.find(".spend-luck");
+			const spendLucknumofdice =  parseInt($(SpendLuck).val());
+			SpendLuck.empty(); // Clear existing options
+			for (let i = 0; i <= this.numOfDice - freedicetoreroll; i++) {
+				const adjustedValue = Math.max(0, i);
+				const selected = i === 0 ? "selected" : "";
+				SpendLuck.append(`<option value="${adjustedValue}" ${selected}>${adjustedValue}</option>`);
+				SpendLuck[0].selectedIndex = spendLucknumofdice;
+			}
+		});
+
+	}
+
+	static async spendluck({actordata, numOfDice, rolltype }) {
```

Comment:
Please use camelCase for function/method names.

Reply:
done